WordPress/gutenberg
# Refactor Toolbar component to TypeScript

Part of #35744

## What?

Convert the `Toolbar` component to TypeScript.

## Why?

As a contribution to the ongoing effort to convert `@WordPress/components` to TypeScript.

## How?

By adding types to `Toolbar`, `@ts-nocheck` where needed for related `Toolbar` components, and refactoring of stories.

## Testing Instructions

1. Create or otherwise open a post
2. Add a block that would have some toolbar controls (e.g. `Image`)
3. Expected: The toolbar should fundamentally work the same as before:
<img width="767" alt="Screen Shot 2023-01-24 at 6 38 05 PM" src="https://user-images.githubusercontent.com/108079556/214454604-606d1c84-8d5b-4296-81ca-31d190e62c56.png">
